File: Data Structures & Algorithms/string-matching-in-an-array/submission-3.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        result: set[str] = set()
        words = sorted(words, key=len)
        for index in range(len(words)):
            for innerIndex in range(index + 1, len(words)):
                if words[innerIndex].find(words[index]) != -1:
                    result.add(words[index])

        return list(result)

File: Data Structures & Algorithms/sliding-window-maximum/submission-3.cpp
class Solution {
public: 
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> result;

        deque<int> sizeIndex; // Sorted decreasing value from index
        for(int currentIndex = 0; currentIndex < nums.size(); currentIndex++){
            int currentNumber = nums[currentIndex];

            while(!sizeIndex.empty() && sizeIndex.front() <= currentIndex - k){
                sizeIndex.pop_front();
            }

            while(!sizeIndex.empty() && currentNumber > nums[sizeIndex.back()]){
                sizeIndex.pop_back();
            }
            
            while(!sizeIndex.empty() && currentNumber > nums[sizeIndex.front()]){
                sizeIndex.pop_front();
            }

            sizeIndex.emplace_back(currentIndex);

            if(currentIndex >= k - 1){
                result.emplace_back(nums[sizeIndex.front()]);
            }
            
        }

        return result;
    }
};



